### DISCUSSION TITLE
Thoughts on using Triplit in a local-only electron app

### DISCUSSION THREAD

**Discussion by harrisjose** [aspen-cloud/triplit#265] [General]
I recently came across Triplit while exploring options for a local-first (SQLite-backed) reactive datastore with relational querying capabilities. Triplit seems like a strong fit for my requirements, but I have a few questions about usage in a local-only Electron app:

- The documentation mentions a local-only mode for the client. Can this mode handle all persistence needs for an app indefinitely, or are there any limitations to be aware of? My guess is that it isn't intended to, as the client and server seem to use different architectures for storing changes.

- I saw on the Lofi Discord that the triplit/db package can be used directly in local-only contexts. If using Triplit without the server is feasible:

  - How would you recommend handling schema migrations and data seeding in this setup?
  - Would it still be possible to use the triplit/client package and the react hooks?

- If the server is something that can't be skipped, maybe one possible approach could involve creating a client that communicates via IPC with a server running in the Electron main process. Does this make sense, or are there better practices for this type of architecture?

Any insights or recommendations would be greatly appreciated. Thank you!

**Comment by matlin** [comment]
I think Triplit could work well for  you. It can be offline for indefinite periods. And with the [Web Worker Client](https://www.triplit.dev/docs/client/web-worker-client) you can share a single DB instance across all browser tabs--which should "just work" with Electron. 

I'd also add if you only need a clientside database, you should also check out RxDB and WatermelonDB. Triplit's features shine brightest when you also have a server that you want to sync your client applications with.
